tools/plugins/spiffe/plugin.go

```diff
@@ -0,0 +1,84 @@
+package spiffe
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/spiffe/go-spiffe/v2/proto/spiffe/workload"
+	"github.com/spiffe/spire/pkg/common/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/apis/clientauthentication"
+	"net"
+	"sigs.k8s.io/cluster-inventory-api/apis/v1alpha1"
+	"sigs.k8s.io/cluster-inventory-api/tools/plugins"
+)
+
+// spiffe plugin requires each spoke cluster to use a single spire server as an oidc provider,
+// and the hub cluster install the spire agent to register all service account to the spire
+// server. Such that the controller on the hub cluster can fetch token from the spire server
+// to be authenticated to the spoke cluster.
+type Plugin struct{}
+
+const pluginName = "spiffe"
+
+type SpiffeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Config Config `json:"config"`
+}
+
+type Config struct {
+	Address  string `json:"address"`
+	Audience string `json:"audience"`
+}
+
+func NewPlugin() (plugins.PluginInterface, error) {
+	return &Plugin{}, nil
+}
+
+func (p *Plugin) Name() string {
+	return pluginName
+}
+
+func (p *Plugin) Credential(cluster *v1alpha1.ClusterProfile) (*clientauthentication.ExecCredential, error) {
```

Comment:
with the latest from the creds kep, you would actually get a "Cluster" object here.
https://github.com/kubernetes/enhancements/pull/5338/files#diff-6dcc0abd0aea71ca2e11e2a5b38ae81f50a227ae1c2c116d058b4de8abd35d20R236

Reply:
I cannot make Cluster in the CRD because of this line https://github.com/kubernetes/kubernetes/blob/a34c07971b610eb33908a743eadb4c61beeecc50/staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1/types.go#L92... 
seems like a bug to me.